examples/loki/nurf_data.py

```diff
@@ -102,120 +102,173 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
+        if data['UV_background'].ndim==1:
+            uv_nb_darks=1
+        else: 
+            uv_nb_darks=np.shape(data['UV_background'])[1]  #TODO: needs to be verified with real data from Judith's setup
+   
+        uv_ik_dark=2* np.ones((1,uv_nb_darks)) 
+        #print(ik_dark)
+        
+        if data['UV_intensity0'].ndim==1:
+            uv_nb_ref=1
+        else:
+            uv_nb_ref=np.shape(data['UV_intensity0'])[1]  #TODO: needs to be verified with real data from Judith's setup
+        uv_ik_ref=4*np.ones((1,uv_nb_ref))  #new image key: 4 for reference
+        #print(ik_ref)
+        
+        # assmebling of image_key
+        uv_image_key=np.column_stack((uv_ik_spectra,uv_ik_dark, uv_ik_ref))  #all_data is as well organised in columns
         
-        #print(list(hf['/entry/'].keys()))
         
-        # create the various subgrous and their attributes
-
         # UV subgroup
         grp_uv = hf.create_group("/entry/instrument/uv")
-        grp_uv.attrs["NX_class"] = 'NXdetector'
-        grp_uv.create_dataset('description', data='UV')
-        grp_uv.create_dataset('type',
-                              data='HR4PRO-UV-VIS-ES, DH-2000-FHS-DUV-TTL, QP600-2-SR')
-    
-        # create the various datasets for UV
-        uv_inttime = grp_uv.create_dataset('uv_integration_time',
-                                           data=data['UV_IntegrationTime'],
-                                           shape=data[
-                                               'UV_IntegrationTime'].shape,
-                                           dtype=np.int32)
-        uv_inttime.attrs['long_name'] = 'uv_integration_time'
-        uv_inttime.attrs['units'] = 's'  # TODO: unit to be verified
-
-        uv_bkg = grp_uv.create_dataset('uv_background',
-                                       data=data['UV_background'],
-                                       shape=data['UV_background'].shape,
-                                       dtype=np.float32)
-        uv_bkg.attrs['long name'] = 'uv_background'
-
-        uv_int0 = grp_uv.create_dataset('uv_intensity0',
-                                        data=data['UV_intensity0'],
-                                        shape=data['UV_intensity0'].shape,
-                                        dtype=np.float32)
-        uv_int0.attrs['long name'] = 'uv_intensity0'
-
-        uv_spectra = grp_uv.create_dataset('uv_spectra',
-                                           data=data['UV_spectra'],
-                                           shape=data['UV_spectra'].shape,
-                                           dtype=np.float32)
-        uv_spectra.attrs['long name'] = 'uv_spectra'
-
-        uv_wavelength = grp_uv.create_dataset('uv_wavelength',
-                                              data=data['UV_wavelength'],
+        grp_uv.attrs["NX_class"] = 'NXdata'
+        
+        # subgroup for uv all data (sample, dark, reference)
+        uv_signal=grp_uv.create_group("uv_all_data")
```

Comment:
What `NX_class` is this? I cannot see where the attribute is set.

Reply:
Thank you. I thought the one level up group `uv` is `NXdata`. I put the `data` and `image_key` into one group. I could get rid of the group and have it as single entries if you prefer this. Otherwise I add to it `NXdata` as well.

Reply:
I don't think nested NXdata is something that is supported. I think the group should just be removed.

Reply:
Done.